iox_query/src/plan/influxql.rs

```diff
@@ -61,58 +104,583 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
+        self.validate_schema_satisfies_exprs(input.schema(), &expr)?;
+        LogicalPlanBuilder::from(input).project(expr)?.build()
+    }
+
+    /// Validate the schema provides all of the columns referenced in the expressions.
+    fn validate_schema_satisfies_exprs(&self, schema: &DFSchema, exprs: &[Expr]) -> Result<()> {
+        find_column_exprs(exprs)
+            .iter()
+            .try_for_each(|col| match col {
+                Expr::Column(col) => match &col.relation {
+                    Some(_r) => {
+                        let arrow: Arc<arrow::datatypes::Schema> = Arc::new(schema.into());
+                        let iox = Schema::try_from(arrow).map_err(|e| {
+                            DataFusionError::Internal(format!("unable to determine schema: {}", e))
+                        })?;
+                        if let Some(i) = iox.find_index_of(col.name.as_str()) {
+                            let (_typ, _f) = iox.field(i);
+                        }
+                        Ok(())
+                    }
+                    None => {
+                        if !schema.fields_with_unqualified_name(&col.name).is_empty() {
+                            Ok(())
+                        } else {
+                            Err(field_not_found(None, col.name.as_str(), schema))
+                        }
+                    }
+                }
+                .map_err(|_: DataFusionError| {
+                    field_not_found(
+                        col.relation.as_ref().map(|s| s.to_owned()),
+                        col.name.as_str(),
+                        schema,
+                    )
+                }),
+                _ => Err(DataFusionError::Internal("Not a column".to_string())),
+            })
+    }
+
+    /// Map the InfluxQL `SELECT` projection list into a list of DataFusion expressions.
+    fn field_list_to_exprs(&self, plan: &LogicalPlan, fields: FieldList) -> Result<Vec<Expr>> {
+        // InfluxQL requires the time column is present in the projection list.
+        let extra = if !has_time_column(&fields) {
+            vec![Field {
+                expr: IQLExpr::VarRef {
+                    name: "time".into(),
+                    data_type: Some(VarRefDataType::Timestamp),
+                },
+                alias: None,
+            }]
+        } else {
+            vec![]
+        };
+
+        extra
+            .iter()
+            .chain(fields.iter())
+            .map(|field| self.field_to_df_expr(field, plan))
+            .collect()
+    }
+
+    /// Map an InfluxQL [`Field`] to a DataFusion [`Expr`].
+    ///
+    /// A [`Field`] is analogous to a column in a SQL `SELECT` projection.
+    fn field_to_df_expr(&self, field: &Field, plan: &LogicalPlan) -> Result<Expr> {
+        let input_schema = plan.schema().as_ref().clone();
+        let expr = self.expr_to_df_expr(ExprScope::Projection, &field.expr, &input_schema)?;
```

Comment:
I think this is copying the entire schema (rather than an Arc). 

```suggestion
        let input_schema = plan.schema();
        let expr = self.expr_to_df_expr(ExprScope::Projection, &field.expr, &input_schema)?;
```

When poking around I also saw the same pattern in `plan_where_clause`